src/main/java/org/takes/facets/fork/MediaType.java

```diff
@@ -108,12 +75,73 @@ public int compareTo(final MediaType type) {
      */
     public boolean matches(final MediaType type) {
         final String star = "*";
-        return (this.high.equals(star)
-            || type.high.equals(star)
-            || this.high.equals(type.high))
-            && (this.low.equals(star)
-            || type.low.equals(star)
-            || this.low.equals(type.low));
+        return (this.highPart().equals(star)
+            || type.highPart().equals(star)
+            || this.highPart().equals(type.highPart()))
+            && (this.lowPart().equals(star)
+            || type.lowPart().equals(star)
+            || this.lowPart().equals(type.lowPart()));
+    }
+
+    /**
+     * Splits the text parts.
+     * @return The parts of the media type.
+     */
+    @Cacheable(forever = true)
```

Comment:
@dalifreire please always address your comments - see http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html#4-avoid-noise-mdash-address-your-comments